src/GraphQL/Resolvers/EventStreamMethodResolver.cs

```diff
@@ -0,0 +1,86 @@
+            : this(methodInfo, methodArgumentExpressions, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes an instance for the specified method, using the specified instance expression to access the instance of the method,
+        /// along with a list of arguments to be passed to the method. The method argument expressions must have return types that match
+        /// those of the method arguments.
+        /// If <paramref name="instanceExpression"/> is <see langword="null"/> then a default instance expression is used as follows:
+        /// <code>context =&gt; (SourceType)context.Source</code>
+        /// </summary>
+        public EventStreamMethodResolver(MethodInfo methodInfo, IList<LambdaExpression> methodArgumentExpressions, LambdaExpression? instanceExpression)
+            : base(methodInfo, methodArgumentExpressions, instanceExpression)
+        {
+        }
+
+        /// <inheritdoc/>
+        protected override Func<IResolveFieldContext, object?> BuildFieldResolver(ParameterExpression resolveFieldContextParameter, Expression bodyExpression)
+        {
+            if (bodyExpression.Type == typeof(Task<IObservable<object?>>))
+            {
+                return Complete(bodyExpression);
+            }
+            else if (bodyExpression.Type.IsGenericType && bodyExpression.Type.GetGenericTypeDefinition() == typeof(Task<>))
+            {
+                var type = bodyExpression.Type.GetGenericArguments()[0];
+                if (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(IObservable<>))
+                {
+                    var innerType = type.GetGenericArguments()[0];
+                    if (!innerType.IsValueType)
+                    {
+                        return Complete(Expression.Call(_castFromTaskAsyncMethodInfo.MakeGenericMethod(innerType), bodyExpression));
+                    }
+                }
+            }
+            else if (bodyExpression.Type.IsGenericType && bodyExpression.Type.GetGenericTypeDefinition() == typeof(IObservable<>))
+            {
+                var innerType = bodyExpression.Type.GetGenericArguments()[0];
+                if (!innerType.IsValueType)
+                {
+                    return Complete(Expression.Call(_castFromObservableMethodInfo.MakeGenericMethod(innerType), bodyExpression));
+                }
+            }
+
+            throw new InvalidOperationException("Method must return a IObservable<T> or Task<IObservable<T>> where T is a reference type.");
+
+            Func<IResolveFieldContext, object?> Complete(Expression bodyExpression)
+            {
+                var lambda = Expression.Lambda<Func<IResolveFieldContext, Task<IObservable<object?>>>>(bodyExpression, resolveFieldContextParameter);
+                _eventStreamResolver = lambda.Compile();
+                return context => context.Source;
```

Comment:
Here we can see the `IFieldResolver` implementation is always `context => context.Source` for a subscription field.  This is correct.